internal/controllers/image/reconcile.go

```diff
@@ -164,11 +164,14 @@ func (r *orcImageReconciler) reconcileNormal(ctx context.Context, orcImage *orcv
 		}
 	}
 
+	log.V(4).Info("Got glance image", "status", glanceImage.Status)
 	log = log.WithValues("ID", glanceImage.ID)
 	ctx = ctrl.LoggerInto(ctx, log)
 
-	log.V(4).Info("Got glance image", "status", glanceImage.Status)
+	return r.handleImageUpload(ctx, imageClient, orcImage, glanceImage, addStatus)
+}
 
+func (r *orcImageReconciler) handleImageUpload(ctx context.Context, imageClient clients.ImageClient, orcImage *orcv1alpha1.Image, glanceImage *images.Image, addStatus func(updateStatusOpt)) (_ ctrl.Result, err error) {
```

Comment:
The name of this function needs some work. This is 'required actions based on the observed status of the resource'.

Good enough for now, and a definite improvement. Lets refine as we add more controllers and get a better feel for it.